pkg/provider/azure_managedDiskController.go

```diff
@@ -211,12 +214,28 @@ func (c *ManagedDiskController) CreateManagedDisk(ctx context.Context, options *
 	}
 
 	cloud := c.common.cloud
-	rerr := cloud.DisksClient.CreateOrUpdate(ctx, options.ResourceGroup, options.DiskName, model)
-	if rerr != nil {
+	disksClient := cloud.DisksClient
+
+	subscriptionID := cloud.subscriptionID
+	if options.SubscrtionID != "" && !strings.EqualFold(options.SubscrtionID, subscriptionID) {
+		// create a new disksClient in cross subscription scenario
+		subscriptionID = options.SubscrtionID
+		if options.ResourceGroup != c.common.resourceGroup {
+			return "", fmt.Errorf("resourceGroup must be specified when subscriptionID(%s) is not empty", subscriptionID)
+		}
+
+		klog.V(2).Infof("azureDisk - create disk(%s) under rg(%s) in subscription(%s)", options.DiskName, options.ResourceGroup, subscriptionID)
+		cloud.ClientConfig.SubscriptionID = subscriptionID
```

Comment:
If this clientConfig is used only here, why not create a local one?

Reply:
I changed the code logic, only add subsID as parameter in snapshot and disk client, PTAL.